resolwe/storage/manager.py

```diff
@@ -32,17 +32,48 @@ def __init__(self, file_storage: FileStorage):
             ("data_slug", data_slug),
         ]
 
-    def _get_rules(self, connector_name: str, _type: str) -> dict:
+    def _get_matching_override_rule(
```

Comment:
This function handles only `data_slug`, and `process_type` overrides, so there is no extra value in `self.override_priorities` as you need to update it for each new type.

I would recommend splitting this function in two, for each override type, and call them explicitly instead of using the `for` loop.